backends/files.go

```diff
@@ -190,119 +204,75 @@ func (o *Files) readAcls() (int, error) {
 		index++
 		line := scanner.Text()
 
-		//Check comment or empty line to skip them.
 		if checkCommentOrEmpty(scanner.Text()) {
 			continue
 		}
 
-		//If we see a user line, change the current user.
-		if strings.Contains(line, "user") {
-			//Try to get username
-			lineArr := strings.Fields(line)
-
-			//Check format
-			if len(lineArr) == 2 && lineArr[0] == "user" {
-				_, ok := o.Users[lineArr[1]]
+		lineArr := strings.Fields(line)
```

Comment:
Due to Fields() we don't support username like `test       with       space`. Because after Fields() + Join() it will result in `test with space`.

Using `strings.Split()` would solve this issue but would introduce issue with line like `topic    read    topic/test/1`, becase `"read"` would not be `lineArr[1]`.

IMO, the two solution we can choose are:
* Don't support username / topic with multiple consecutive space. But from my understanding Mosquitto support this and this is requested (https://github.com/iegomez/mosquitto-go-auth/issues/154#issuecomment-783213816)
* Use smarter Split/Fields. I think writing some kind for `FieldsN()` that split/yield up to Nth fields with last being unsplitted is doable. We should then use FieldsN(line, 2) for user line and FieldsN(line, 3) for topic.

Ideally we should use a tokenizer (like Mosquitto does with strtok), from which we could ask the next fields or the rest of line. But such tokenizer seems more complex than writing FieldsN which seems enough (since we have only 2 kind of line: user or topic/pattern).

Reply:
The FieldsN I though is something like this: https://gist.github.com/PierreF/3866b140309b67e4507e4a52c691c353

Reply:
@PierreF Maybe it's github's formatting that's taking out some spaces, because both usernames look the same and the topic only shows one space between `topic` and `read`, and I believe you were showing that 2 spaces in between wouldn't rot, right?
I don't think is difficult to process it, but should we do that? The correct format is `topic read ...`, i.e. with one space, as far as I know.

Reply:
I may be wrong about that last statement though.

Reply:
Never mind, just checked Mosquitto's code and it indeed allows it.

Reply:
I've updated my original comment. Didn't see github didn't allowed multiple space in `single quoted value`.